File: Julia/8 - Newton.jl
function f(x)
    return x^4+2*x^3-13*x^2-14*x+24
end

function dff(x)
    return 4*x^3+6*x^2-26*x-14
end

function Newton()
    x0 = 4
    toler = 1.0000e-5
    iterMax = 100
    fx = f(x0)
    dFx = dff(x0)
    x = x0
    iter = 0
    deltaX = 1000
    df = 1
    condErro = 0;
    h = abs(deltaX)
    println("deltaX ====> $h")
    j = abs(fx)
    println("fx ====> $j")
    while (abs(deltaX)>toler || abs(fx)>toler) && df==1  && iter<iterMax
        deltaX = fx/dFx
        x = x + deltaX
        fx = f(x)
        dFx = dff(x)
        iter = iter + 1
    end
    raiz = x;
    if abs(deltaX)<=toler && abs(fx)<=toler
        condErro = 0
    else
        condErro = 1;
    println("CondErro ====> $condErro")
    println("Raiz ====> $raiz")
    println("Iter ====> $iter")
    end
end

println("TEMPO: ",@elapsed Newton())

File: C/8 - Newton.c
#include <stdio.h>
#include <math.h>

double ff(double x) {
    return pow(x,4)+2*pow(x,3)-13*pow(x,2)-14*x+24;
}

double dff(double x) {
    return 4*pow(x,3)+6*pow(x,2) -26*x-14;
}

void Newton () {
    int x0 = 4, iterMax = 100;
    double toler = 1.0000e-05, y, x;
    double fx, dfx, deltax, df, raiz;
    int iter = 0, condErro;
    fx = ff(x0);
    dfx = dff(x0);
    x = x0;
    deltax = 1000;
    df = 1;

    while((fabs(deltax)>toler || fabs(fx)>toler) && df!=0 && iter<iterMax) {
        deltax = fx/dfx;
        x = x + deltax;
        fx = ff(x);
        dfx = dff(x);
        iter++;
    }

    raiz = x;

    if (fabs(deltax)<=toler && abs(fx)<=toler) {
        condErro = 0;
    }
    else {
        condErro = 1;
    }
    printf("Raiz ===> %f\n", raiz);
    printf("condErro ===> %d \n", condErro);
    printf("iter ===> %d \n\n\n", iter);
}

int main() {
    Newton();
    return 0;
}
